src/compiler/checker.ts

```diff
@@ -12958,6 +12961,30 @@ namespace ts {
                                 return Ternary.False;
                             }
                         }
+                        else {

```

Comment:
This location seems weird since we later check index signatures in indexTypeRelatedTo. It might just be because we've never tried this kind of check before. But it might also be the source of the odd new errors in the user tests and RWC tests. I'll take a closer look at those failures now.

Reply:
`indexTypesRelatedTo` ensures that if the _target_ has an index signature, then the source either has a matching index signature or satisfies the special rule for object literal types or generic mapped types.  This code is doing something different: ensuring that an optional property of the target that has no matching source property doesn't receive an incompatible value from an index signature of the _source_.  Since the check has to be done for each unmatched property of the target, here seemed like the logical place to put it.